### PR TITLE
Update data docs

### PR DESCRIPTION
**Pull request by nd-02110114**
## Description

I merged `datasets`, `dataloaders` and `dataclasses` to `data.rst`. This is because it is difficult for new comers to understand the difference among them when the new comers see them in the sidebar of documents.

In addition to this, I fixed some docstring which is not displayed correctly.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/data/datasets.py [hunk 1]
```diff
@@ -1052,24 +1052,24 @@ class DiskDataset(Dataset):
...
+  |   ---> shard-0-y.npy
+  |   |
+  |   ---> shard-0-w.npy
+  |   |
+  |   ---> shard-0-ids.npy
+  |   |
+  |   ---> shard-1-X.npy
+  |   .
+  |   .
+  |   .
```
Comment (rbharath) [inline deepchem/data/datasets.py:1072]: Ah nice fix!

File: deepchem/data/datasets.py [hunk 2]
```diff
@@ -1371,7 +1371,7 @@ def move(self, new_data_dir: str,
       before moving. This is set to True by default to be backwards compatible
       with behavior in earlier versions of DeepChem.
 
-    Notes
+    Note
     -----
```
Comment (rbharath) [inline deepchem/data/datasets.py:1375]: I think you have an extra `_` here and in a couple places below

**Review by rbharath — APPROVED** [review]
I think this simplification looks good to me! Feel free to merge in once ready

File: deepchem/data/datasets.py (hunk 2 above)
Comment (nd-02110114) [reply deepchem/data/datasets.py]: I fixed!

**Comment by nd-02110114** [comment]
I merge in.

**Pull request merged**